File: cci05.07-exchange-lcci/python/Solution.py
class Solution:
    def exchangeBits(self, num: int) -> int:
        ret = 0
        for i in range(0, 30, 2):
            a1 = (num & (1 << i))
            b1 = (num & (1 << (i+1)))
            if a1 != 0:
                ret |= (1<<(i+1))
            if b1 != 0:
                ret |= (1<<i)
        return ret

File: cci05.07-exchange-lcci/cpp/Solution.cpp
class Solution {
public:
    int exchangeBits(int num) {
        int ret = 0;
        for (int i = 0; i <= 30; i += 2) {
            int a1 = (num & (1 << i));
            int b1 = (num & (1 << (i + 1)));
            if (a1 != 0) ret |= (1 << (i + 1));
            if (b1 != 0) ret |= (1 << i);
        }
        return ret;
    }
};
